app/assets/javascripts/lookup_keys.js

```diff
@@ -7,8 +7,21 @@ $(function() {
   $('.tabs-left .col-md-4').removeClass('col-md-4').addClass('col-md-8')
   //remove variable click event
   $(document).on('click', '.smart-var-tabs li a span', function(){ remove_node(this);});
+  fill_in_matchers();
+  $('#smart_class_param').parents('form').on('submit', function(){
+    build_match();
+  })
+  $('.matcher_key').select2('destroy');
+  $('#optional_input_validators').on('hidden.bs.collapse', toggleCaret);
+  $('#optional_input_validators').on('shown.bs.collapse', toggleCaret);
 })
 
+function toggleCaret(e) {
+  $(e.target)
+    .parent()
+    .toggleClass('dropup ');
```

Comment:
As mentioned over IRC, you could remove all of this toggle JS by extending the bootstrap 3 class.  Something like this (untested):

```scss
@import 'bootstrap';
 
.collapsed .caret {
  @extend .dropup .caret;
}
```